pybop/models/base_model.py

```diff
@@ -238,37 +246,36 @@ def set_initial_state(self, initial_state: dict):
         # Use a copy of the updated default parameter set
         self._parameter_set = self._unprocessed_parameter_set.copy()
 
-    def set_params(self, rebuild: bool = False, dataset: Dataset = None):
+    def set_current_function(self, dataset: Union[Dataset, dict]):
+        """
+        Update the input current function according to the data.
+
+        Parameters
+        ----------
+        dataset : pybop.Dataset or dict, optional
+            The dataset to be used in the model construction.
+        """
+        if "Current function [A]" in self._parameter_set.keys():
+            self._parameter_set["Current function [A]"] = pybamm.Interpolant(
```

Comment:
It looks like the line below has been removed. Was this catch added elsewhere? It's functionality is for users wanting to fit a stationary current function from provided data, i.e. mark the current function as `"input"`.

```python
if "Current function [A]" not in self.parameters.keys():
```

Reply:
In this case, I don't think it would be poassible to pass a dataset?

Reply:
To be sure, I've added the check back in